### DISCUSSION TITLE
RE: "GUI: Bring Toggl reports to code blocks"

### DISCUSSION THREAD

**Discussion by jebeaudoin** [mcndt/obsidian-toggl-integration#26] [Feedback]
Before I make a feature request, I wanted to make sure I understood this roadmap item. If implemented, would this enable a user to display, for instance, all the Toggl entries for a specific date in an Obisidian note? If so, I'd strongly +1 this feature.

I would love to include a Toggl code block in my daily and weekly review notes. Thanks!
Jack

**Comment by mcndt** [comment]
That's exactly the kind of use case I had in mind with this roadmap item. The exact implementation (what data can be queried, how it is visualized) I have not thought about yet because there are some more pressing missing features that need to be done first (mainly tagging).

Feel free to explain how you would like to use Toggl reports in your daily/weekly review notes! Your feedback can help shape the feature when it is eventually implemented.

**Comment by jebeaudoin** [comment]
Awesome...

The daily report code block is straightforward for me. Just like you'd see under "Today" at https://track.toggl.com/timer 
Time entry // Project // Tag(s) // Start Time // End Time // Duration

The weekly report code block would sum durations by any of project, tag or client. Since I'm a solo freelancer, the "user" entity in Toggl Track isn't of much value to me.

I suppose that instead of hard coding a weekly code block, it would be more beneficial to set a start and end parameter. That way, the end user could create weekly, monthly, quarterly or annual summaries.

Thanks for the development work. If I adopt a plugin, I do contribute to the developer because I understand how busy you folks are and I appreciate the work!
